### PR TITLE
[action] add skip_info_plist parameter to increment_build_number to avoid updating Info.plist

### LINKED ISSUE #18367: increment_build_number should not touch CFBundleVersion if CURRENT_PROJECT_VERSION
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
`increment_build_number` currently updates `CURRENT_PROJECT_VERSION` in `xcodeproj` as well `CFBundleVersion` in `Info.plist`.

This is undesirable when `Info.plist`'s `CFBundleVersion` is already set to `$(CURRENT_PROJECT_VERSION)`. The change just creates noise in the repo and has no effect. The `Info.plist` is already set up to source its value from the already-modified `CURRENT_PROJECT_VERSION`.

##### Command executed
```
bundle exec fastlane run increment_build_number
```

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 11.2.2                                                         |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.30.1                                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | macOS 11.2.2 (20D80)                                           |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 12.4                                                           |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_CA.UTF-8 | ✅ |
| LC_ALL   | en_CA.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
opt_out_usage
update_fastlane

platform :ios do
  desc "Build an application archive"
  lane :archive do
    ensure_no_debug_code(text: "<<<<<<|WIP:", exclude_dirs: [".git", "fastlane"])
    #verify_xcode
    cocoapods
    #swiftlint

    build_app(scheme: "Spectre-iOS", export_options: { method: "ad-hoc" })
    verify_build
  end

  desc "Build an application archive and send it to TestFlight"
  lane :beta do
    ensure_git_status_clean
    ensure_git_branch(branch: "^stream/(alpha|beta|probe)$")

    archive
    install_on_device(skip_wifi: true)
    prompt(boolean: true, text: "Please test the app on your device. Is everything alright?") # TODO: check
    upload_to_testflight
    download_dsyms
    sentry_upload_dsym #(include_sources: true)

    build = get_build_number().split(".").map(&:to_i)
    build[-1] += 1
    increment_build_number(build_number: build.join("."))
  end

  lane :release do
    #precheck
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("app.spectre")

apple_id("lhunath@lyndir.com")
itc_team_id("153897")
team_id("HL3Q45LX9N")

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.177.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                 | Version | Update-Status |
| ---------------------- | ------- | ------------- |
| fastlane-plugin-sentry | 1.8.1   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.5.0        |
| slack-notifier                | 2.3.2        |
| rouge                         | 2.0.7        |
| xcpretty                      | 0.3.0        |
| terminal-notifier             | 2.0.0        |
| terminal-table                | 1.8.0        |
| plist                         | 3.6.0        |
| addressable                   | 2.7.0        |
| multipart-post                | 2.0.0        |
| word_wrap                     | 1.0.0        |
| tty-screen                    | 0.8.1        |
| tty-cursor                    | 0.7.1        |
| tty-spinner                   | 0.9.3        |
| artifactory                   | 3.0.15       |
| babosa                        | 1.0.4        |
| colored                       | 1.2          |
| highline                      | 1.7.10       |
| commander-fastlane            | 4.4.6        |
| excon                         | 0.79.0       |
| faraday-net_http              | 1.0.1        |
| ruby2_keywords                | 0.0.4        |
| faraday                       | 1.3.0        |
| unf_ext                       | 0.0.7.7      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20190701 |
| http-cookie                   | 1.0.3        |
| faraday-cookie_jar            | 0.0.7        |
| faraday_middleware            | 1.0.0        |
| fastimage                     | 2.2.3        |
| gh_inspector                  | 1.1.3        |
| mini_magick                   | 4.11.0       |
| naturally                     | 2.2.1        |
| rubyzip                       | 2.3.0        |
| security                      | 0.1.3        |
| xcpretty-travis-formatter     | 1.0.1        |
| dotenv                        | 2.7.6        |
| simctl                        | 1.6.8        |
| jwt                           | 2.2.2        |
| uber                          | 0.1.0        |
| declarative                   | 0.0.20       |
| declarative-option            | 0.1.0        |
| representable                 | 3.0.4        |
| retriable                     | 3.1.2        |
| mini_mime                     | 1.0.2        |
| httpclient                    | 2.8.3        |
| google-api-client             | 0.38.0       |
| google-cloud-errors           | 1.0.1        |
| google-cloud-core             | 1.5.0        |
| digest-crc                    | 0.6.3        |
| google-cloud-storage          | 1.30.0       |
| emoji_regex                   | 3.2.2        |
| nanaimo                       | 0.3.0        |
| colored2                      | 3.1.2        |
| claide                        | 1.0.3        |
| CFPropertyList                | 3.0.3        |
| atomos                        | 0.1.3        |
| xcodeproj                     | 1.19.0       |
| unicode-display_width         | 1.7.0        |
| public_suffix                 | 4.0.6        |
| json                          | 2.5.1        |
| bundler                       | 2.2.7        |
| multi_json                    | 1.15.0       |
| signet                        | 0.15.0       |
| os                            | 1.1.1        |
| memoist                       | 0.16.2       |
| googleauth                    | 0.16.0       |
| google-cloud-env              | 1.5.0        |
| webrick                       | 1.7.0        |
| rexml                         | 3.2.4        |
| google-apis-core              | 0.3.0        |
| google-apis-iamcredentials_v1 | 0.2.0        |
| google-apis-storage_v1        | 0.3.0        |
| rake                          | 13.0.3       |
| aws-eventstream               | 1.1.1        |
| aws-sigv4                     | 1.2.3        |
| aws-partitions                | 1.431.1      |
| jmespath                      | 1.4.0        |
| aws-sdk-core                  | 3.112.1      |
| aws-sdk-kms                   | 1.42.0       |
| aws-sdk-s3                    | 1.90.0       |
| forwardable                   | 1.3.2        |
| logger                        | 1.4.3        |
| date                          | 3.1.1        |
| stringio                      | 3.0.0        |
| zlib                          | 1.1.0        |
| ipaddr                        | 1.2.2        |
| openssl                       | 2.1.2        |
| ostruct                       | 0.3.3        |
| strscan                       | 3.0.0        |
| fileutils                     | 1.5.0        |
| io-console                    | 0.5.7        |
| libxml-ruby                   | 3.2.1        |
| psych                         | 3.3.0        |
| mutex_m                       | 0.1.1        |
| ox                            | 2.13.2       |
| fastlane-plugin-sentry        | 1.8.1        |
</details>

*generated on:* **2021-03-09**
</details>

### PR DESCRIPTION
**Pull request by lhunath**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
This is important when Info.plist is configured to reference $(CURRENT_PROJECT_VERSION).

Fixes #18367

### Description
The issue is caused by passing `-all` to `agvtool`. This change makes `-all` optional by allowing the developer to turn it off.

### CODE REVIEW

File: fastlane/lib/fastlane/actions/increment_build_number.rb [hunk 1]
```diff
@@ -66,6 +67,11 @@ def self.available_options
                                        description: "Change to a specific version. When you provide this parameter, Apple Generic Versioning does not have to be enabled",
                                        optional: true,
                                        is_string: false),
+          FastlaneCore::ConfigItem.new(key: :skip_info,
+                                       env_name: "FL_BUILD_NUMBER_SKIP_INFO",
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/increment_build_number.rb:71]:
```suggestion
          FastlaneCore::ConfigItem.new(key: :skip_info_plist,
                                       env_name: "FL_BUILD_NUMBER_SKIP_INFO_PLIST",
```

File: fastlane/lib/fastlane/actions/increment_build_number.rb [hunk 2]
```diff
@@ -33,10 +33,11 @@ def self.run(params)
         agv_disabled = !system([command_prefix, 'agvtool what-version', command_suffix].join(' '))
         raise "Apple Generic Versioning is not enabled." if agv_disabled && params[:build_number].nil?
 
+        mode = params[:skip_info] ? '' : '-all'
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/increment_build_number.rb:36]:
```suggestion
        mode = params[:skip_info_plist] ? '' : '-all'
```

**Review by joshdholtz — CHANGES_REQUESTED** [review]
One small nit pick in the naming 😊 This PR look 💯 other than this! ❤️

File: fastlane/lib/fastlane/actions/increment_build_number.rb [hunk 3]
```diff
@@ -66,6 +67,11 @@ def self.available_options
                                        description: "Change to a specific version. When you provide this parameter, Apple Generic Versioning does not have to be enabled",
                                        optional: true,
                                        is_string: false),
+          FastlaneCore::ConfigItem.new(key: :skip_info,
+                                       env_name: "FL_BUILD_NUMBER_SKIP_INFO",
+                                       description: "Don't update Info.plist files when updating the build version",
+                                       is_string: false,
+                                       default_value: false),
```
Comment (minuscorp) [inline fastlane/lib/fastlane/actions/increment_build_number.rb:74]: Add `type: Boolean` for a nice parsing into Swift 🙏

**Comment by joshdholtz** [comment]
@googlebot I consent.

**Review by joshdholtz — APPROVED** [review]
Thanks for adding this feature in! I added a commit in here to add some tests but other than that this was 💯 Really appreciate the contribution ❤️

**Comment by Dinsen** [comment]
@joshdholtz remember to change the title to contain the key that was updated doing the review `add skip_info_plist..` so release notes are correct 🥰

**Comment by joshdholtz** [comment]
@Dinsen Thanks for reminder!

**Pull request merged**